config/docker/rustc-1.62.jinja2

```diff
@@ -0,0 +1,42 @@
+{% extends 'clang-15.jinja2' %}
+
+{% block packages %}
+{{ super() }}
+
+ENV RUST_VER=1.62.0
+ENV BINDGEN_VER=0.56.0
```

Comment:
It's possible to set build configurations with kernel version filters to only build particular kernel versions with particular toolchain versions.  However this will require manual maintenance in the YAML configuration.

Reply:
Yeah, I imagined something like that. Then it is perfectly fine creating sets of build images :)

> We could add a test in KernelCI for the kernel builds which verifies what specific Rust toolchain version is required and fail the build if there is a toolchain version mismatch, though I don't know if that makes sense,

The kernel already checks if the version is recent enough, so it should be fine. It also checks if it is too old, but that is a warning only.

> depending on what the plans are for Rust toolchain backwards compatibility in the kernel? Maybe it's still too early to know this?

The goal is to provide a minimum Rust supported version like for GCC and Clang (that can be upgraded from time to time like any other), but before that happens some unstable features need to be stabilized on the Rust side (or at least "semi-stabilized", even if not fully stable for general use).

Reply:
> It's possible to set build configurations with kernel version filters to only build particular kernel versions with particular toolchain versions. However this will require manual maintenance in the YAML configuration.

Sounds good to me (or do you mean this is an alternative vs. something else?)

How should we coordinate the changes? i.e. if we submit an upgrade to mainline, should we submit it here before Linus merges the PR? (setting it to the next release? Or per -rc?)

Reply:
The kernel already checks the version, so I assume you will get a fairly clear build error if you are using the lowest version previously supported before a given upgrade. But yeah, adding an specific test sounds good too if that gives better reports.

As for intervention from the kernel maintainers side, @gctucker asked us in the mailing list whether we would help with this, so I assumed we would have to submit the updates or something like that. I would be happy to do so if needed; but if things are more streamlined, even better!